src/pages/keyword-generator/KeywordGenerator.vue

```diff
@@ -1,9 +1,71 @@
 <script setup>
+import { ref } from 'vue'
+
+const sentence = ref('')
+const maxN = ref(3)
+const combinations = ref([])
+
+const handle = (event) => {
+  event.preventDefault()
+
+  // Reset combinations
+  combinations.value = []
+
+  const words = sentence.value.split(/\s+/)
+
+  for (let n = 1; n <= maxN.value; n++) {
+    const uniqueCombinations = new Set()
+
+    for (let i = 0; i <= words.length - n; i++) {
+      const combination = words.slice(i, i + n).join(' ')
+      uniqueCombinations.add(combination)
+    }
+
+    if (uniqueCombinations.size != 0) {
+      combinations.value.push(Array.from(uniqueCombinations).join(", "))
+    }
+  }
+}
 </script>
+
 <template>
     <div class="ma-keywords-generator">
-        <div class="ma-header">
-            <span>Keyword Generator </span>
-        </div>
+        <form>
+            <div class="ma-header">
+                <label class="text-black-500 text-3xl" for="allwords">Keyword Generator</label>
+            </div>
+            <div>
+                <textarea class="input-area" id="allwords" v-model="sentence"></textarea>
+            </div>
+            <div>
+                <button style="margin-top: 10px;" @click="handle">Submit</button>
+                <div v-for="(combo, index) in combinations" :key="index" class="preview">
```

Comment:
```suggestion
                <div v-for="(combination, index) in combinations" :key="index" class="preview">
```

We should avoid abbreviating variable names. If we need to give a long name to describe what the variable does, we should do so.